src/app/pages/quickMaps/pages/projection/projectionFoodSources/projectionFoodSources.component.ts

```diff
@@ -6,28 +7,83 @@ import {
 import { CardComponent } from 'src/app/components/card/card.component';
 import { CurrentDataService } from 'src/app/services/currentData.service';
-import { DialogService } from 'src/app/components/dialogs/dialog.service';
 import { MAT_DIALOG_DATA } from '@angular/material/dialog';
 import { DialogData } from 'src/app/components/dialogs/baseDialogService.abstract';
+import {
+  ProjectedFoodSourcesData,
+  ProjectedFoodSourcesPeriod,
+  ProjectedFoodSourcesTable,
+} from 'src/app/apiAndObjects/objects/projectedFoodSources';
+import { ChartJSObject } from 'src/app/apiAndObjects/objects/misc/chartjsObject';
+import { MatTableDataSource } from '@angular/material/table';
+import { MatTabGroup } from '@angular/material/tabs';
+import { MatSort } from '@angular/material/sort';
+import { MiscApiService } from 'src/app/services/miscApi.service';
+import { ImpactScenario } from 'src/app/apiAndObjects/objects/impactScenario';
+import { FormBuilder, FormGroup } from '@angular/forms';
+import { Dictionary } from 'src/app/apiAndObjects/_lib_code/objects/dictionary';
+import { DictionaryService } from 'src/app/services/dictionary.service';
 
 @Component({
   selector: 'app-proj-food-sources ',
   templateUrl: './projectionFoodSources.component.html',
-  styleUrls: [
-    '../../expandableTabGroup.scss',
-    './projectionFoodSources.component.scss',
-  ],
+  styleUrls: ['../../expandableTabGroup.scss', './projectionFoodSources.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class ProjectionFoodSourcesComponent implements AfterViewInit {
+export class ProjectionFoodSourcesComponent implements OnInit, AfterViewInit {
+  @ViewChild(MatTabGroup) tabGroup: MatTabGroup;
+  @ViewChild(MatSort) set matSort(ms: MatSort) {
+    this.sort = ms;
+    if (this.dataSource) {
+      this.dataSource.sort = this.sort;
+    }
+  }
+
   @Input() card: CardComponent;
 
-  public title = 'Projection Food Sources';
+  public title = 'Projection Food Sources for';
+  public micronutrientsDictionary: Dictionary;
+  public micronutrientName = '';
+  public chartData: ChartJSObject;
+  public displayedColumns = ['foodName', 'value'];
+  public dataSource = new MatTableDataSource();
+  public selectedCountry: string;
+  public currentImpactScenario: ImpactScenario;
+  public projectionFoodFormGroup: FormGroup;
+  public groupByOptions = [
+    { id: 'commodity', name: 'commodity' },
+    { id: 'foodGroup', name: 'food group' },
+  ];
+  public scenarioOptions = [
+    { id: 'SSP1', name: 'SSP1' },
+    { id: 'SSP2', name: 'SSP2' },
+    { id: 'SSP3', name: 'SSP3' },
+  ];
+  public yearOptions = [
+    { id: 0, name: '2005' },
+    { id: 1, name: '2010' },
+    { id: 2, name: '2015' },
+    { id: 3, name: '2020' },
+    { id: 4, name: '2025' },
+    { id: 5, name: '2030' },
+    { id: 6, name: '2035' },
+    { id: 7, name: '2040' },
+    { id: 8, name: '2045' },
+    { id: 9, name: '2050' },
+  ];
+
+  public currentYearId: number = 0;
```

Comment:
are you talking about the groupByOptions, scenarioOptions and yearOptions?

Reply:
The "currentYearId" variable, seems to be superfluous.

Looked like when you wanted to use it you could just reference the value from the control instead of setting a local variable every time it changed.

Reply:
in which case fixed :)